eng/common/TestResources/New-TestResources.ps1

```diff
@@ -155,6 +157,93 @@ function BuildBicepFile([System.IO.FileSystemInfo] $file)
     return $templateFilePath
 }
 
+function BuildDeploymentOutputs([string]$serviceDirectoryPrefix, [object]$azContext, [object]$deployment) {
+    # Add default values
+    $deploymentOutputs = [Ordered]@{
```

Comment:
Wonder if we keep configuration next to the values for the build-in ones.

Reply:
If I understand you correctly, we could but it also varies per cloud. I initially opted to put them in separate json files alongside this script to make it easier to communicate to other developers what the per-cloud values are by linking to files/file naming convention.

Reply:
What I'm trying to say is there is a list of well-known variables defined in the New-TestResources script. It's unfortunate that we have to go to a completely different place to state that they are not secrets.

Reply:
And to also have the logic to reverse-engineer the prefixing rules there.

Reply:
Can we populate ShouldMarkValueAsSecret from BuildDeploymentOutputs directly?

Reply:
It would certainly be cleaner. There are a couple other concerns which is why I went with this approach over just updating the data structure, but I'd certainly appreciate ideas. Sorry this is a bunch of text:

1. In order to support merging multiple subscription configurations (for example in the case where a partner team needs to use their own subscription, but still authorize with our service principal, or they need language or cloud-specific overrides), there is a [pre-requisite step](https://github.com/Azure/azure-sdk-tools/blob/b03f41a2f98778899bf1943834a05fc9288ce8e0/eng/common/TestResources/build-test-resource-config.yml) that runs before deployment in the live test pipelines to download and merge these configs. This step also uses the `ShouldMarkValueAsSecret` code to avoid marking certain values as secret. I was finding once we set them as secret once, we could not unset them, but I was testing a couple other things at the same time so I need to verify in isolation. We can't know ahead of time what the values or prefixes might be, or where potential duplicate values with different keys are set, so I tried to address this by taking a more general naming convention approach.
2. Similar to 1. to support a secret marking design that is entirely set in the data structure would also require updating all the existing keyvault secrets that we and partner teams own OR having some sort of backwards compatible approach where people could opt-in to mark things as non-secret if they wanted. Initially I thought that this was more work than it was worth but it's an approach we could take.
3. I don't anticipate these values changing or being updated very frequently, so having the more cumbersome list stored in multiple places seemed like less of a downside than solving for the above two problems.

I think the prefixing strategy/convention is something that's probably worth talking about. I don't see much value in maintaining the service directory variable prefixing right now and am considering what it would mean to just use `AZURE_` everywhere.

Reply:
Thank you for the thoughtful answer ❤️

>I think the prefixing strategy/convention is something that's probably worth talking about.

The original hope was that we can deploy multiple sets of test resources and they won't conflict with each other because of the prefix.